File: test/compiler/change_detector_mocks.dart
library angular2.test.compiler.change_detector_mocks;

import "package:angular2/src/facade/lang.dart" show isBlank;
import "package:angular2/src/core/change_detection/pipes.dart" show Pipes;
import "package:angular2/src/facade/async.dart" show EventEmitter;
import "package:angular2/src/core/change_detection/change_detection.dart"
    show ChangeDetector, ChangeDispatcher, DirectiveIndex, BindingTarget;

class TestDirective {
  List<String> eventLog = [];
  String dirProp;
  EventEmitter<dynamic> click = new EventEmitter<dynamic>();
  onEvent(String value) {
    this.eventLog.add(value);
  }
}

class TestDispatcher implements ChangeDispatcher {
  List<dynamic> directives;
  List<ChangeDetector> detectors;
  List<String> log;
  TestDispatcher(this.directives, this.detectors) {
    this.clear();
  }
  getDirectiveFor(DirectiveIndex di) {
    return this.directives[di.directiveIndex];
  }

  getDetectorFor(DirectiveIndex di) {
    return this.detectors[di.directiveIndex];
  }

  clear() {
    this.log = [];
  }

  notifyOnBinding(BindingTarget target, value) {
    this.log.add(
        '''${ target . mode}(${ target . name})=${ this . _asString ( value )}''');
  }

  logBindingUpdate(target, value) {}
  notifyAfterContentChecked() {}
  notifyAfterViewChecked() {}
  notifyOnDestroy() {}
  getDebugContext(a, b, c) {
    return null;
  }

  _asString(value) {
    return (isBlank(value) ? "null" : value.toString());
  }
}

class TestPipes implements Pipes {
  get(String type) {
    return null;
  }
}

File: modules/angular2/test/compiler/change_detector_mocks.ts
import {isBlank} from 'angular2/src/facade/lang';
import {Pipes} from 'angular2/src/core/change_detection/pipes';
import {EventEmitter} from 'angular2/src/facade/async';
import {
  ChangeDetector,
  ChangeDispatcher,
  DirectiveIndex,
  BindingTarget
} from 'angular2/src/core/change_detection/change_detection';

export class TestDirective {
  eventLog: string[] = [];
  dirProp: string;
  click: EventEmitter<any> = new EventEmitter<any>();

  onEvent(value: string) { this.eventLog.push(value); }
}

export class TestDispatcher implements ChangeDispatcher {
  log: string[];

  constructor(public directives: any[], public detectors: ChangeDetector[]) { this.clear(); }

  getDirectiveFor(di: DirectiveIndex) { return this.directives[di.directiveIndex]; }

  getDetectorFor(di: DirectiveIndex) { return this.detectors[di.directiveIndex]; }

  clear() { this.log = []; }

  notifyOnBinding(target: BindingTarget, value) {
    this.log.push(`${target.mode}(${target.name})=${this._asString(value)}`);
  }

  logBindingUpdate(target, value) {}

  notifyAfterContentChecked() {}
  notifyAfterViewChecked() {}

  notifyOnDestroy() {}

  getDebugContext(a, b, c) { return null; }

  _asString(value) { return (isBlank(value) ? 'null' : value.toString()); }
}

export class TestPipes implements Pipes {
  get(type: string) { return null; }
}
